.pre-commit-config.yaml

```diff
@@ -45,3 +44,36 @@ repos:
 #   rev: v1.3.0
 #   hooks:
 #     - id: dead
+- repo: https://github.com/jumanjihouse/pre-commit-hooks
+  rev: 1.11.0
+  hooks:
+    - id: script-must-have-extension
+      types: [shell, non-executable]
+    - id: shellcheck
+      args: [--color=always, --check-sourced, --external-sources]
+      additional_dependencies: [shellcheck]
+    - id: shfmt
+      additional_dependencies: [shfmt]
```

Comment:
This hook uses shellcheck and shfmt to find potential scripting errors. Not sure what happens if user doesn't have them installed, may just ignore.